x/globalfee/module.go

```diff
@@ -0,0 +1,142 @@
+package globalfee
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	"github.com/gorilla/mux"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/bandprotocol/chain/v2/x/globalfee/client/cli"
+	"github.com/bandprotocol/chain/v2/x/globalfee/types"
+)
+
+var (
+	_ module.AppModuleBasic   = AppModuleBasic{}
+	_ module.AppModuleGenesis = AppModule{}
+	_ module.AppModule        = AppModule{}
+)
+
+// AppModuleBasic defines the basic application module used by the wasm module.
+type AppModuleBasic struct{}
+
+func (a AppModuleBasic) Name() string {
+	return types.ModuleName
+}
+
+func (a AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
+	return cdc.MustMarshalJSON(&types.GenesisState{
+		Params: types.DefaultParams(),
+	})
+}
+
+func (a AppModuleBasic) ValidateGenesis(
+	marshaler codec.JSONCodec,
+	config client.TxEncodingConfig,
+	message json.RawMessage,
+) error {
+	var data types.GenesisState
+	err := marshaler.UnmarshalJSON(message, &data)
+	if err != nil {
+		return err
+	}
+	if err := data.Params.ValidateBasic(); err != nil {
+		return sdkerrors.Wrap(err, "params")
+	}
+	return nil
+}
+
+func (a AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+}
+
+func (a AppModuleBasic) RegisterRESTRoutes(context client.Context, router *mux.Router) {
+}
```

Comment:
I thought 0.46 don't use this function anymore?

Reply:
ummm, let me check

Reply:
removed.